src/main/java/org/bonitasoft/connectors/ws/SecureWSConnector.java

```diff
@@ -297,36 +296,16 @@ private void sanitizeInputs() {
         sanitizedInputs.put(PROXY_PROTOCOL, sanitizeString((String) getInputParameter(PROXY_PROTOCOL)));
         sanitizedInputs.put(PROXY_USER, sanitizeString((String) getInputParameter(PROXY_USER)));
         sanitizedInputs.put(PROXY_PASSWORD, sanitizeString((String) getInputParameter(PROXY_PASSWORD)));
-        //Booleans, no need to sanitize them
+        // No need to sanitize the HTTP Headers
+        sanitizedInputs.put(HTTP_HEADERS, getInputParameter(HTTP_HEADERS));
```

Comment:
i think we need to test this, before avoid escaping header. Because it look like most servers refuse to take any non-printable chars in headers.
https://stackoverflow.com/questions/47687379/what-characters-are-allowed-in-http-header-values